src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -552,15 +552,34 @@ protected function processParams(File $phpcsFile, $stackPtr, $commentStart)
             // Make sure the param name is correct.
             if (isset($realParams[$pos]) === true) {
                 $realName = $realParams[$pos]['name'];
-                if ($realName !== $param['var']) {
+
+                if ($realParams[$pos]['pass_by_reference'] === true && $param['var'][0] === '&') {
+                    // When passed by reference, the variable name in $realParams does not
+                    // have a leading '&'. There is no guidance in PSR-19 to say if the
+                    // docblock SHOULD or SHOULD NOT also have a leading '&'. Therefore,
+                    // this sniff remains agnostic in this case.
+                    $paramVarName = substr($param['var'], 1);
+
+                    // This fixes a false-positive with 'MissingParamTag'.
+                    $foundParams[(count($foundParams) - 1)] = $paramVarName;
+                } else {
+                    // When not passed by reference, the name should match exactly. If
+                    // there is a stray leading '&', then something has gone wrong and
+                    // either the docblock or the function / method signature needs to be
+                    // changed. We can't auto-fix this as it's not clear which of these
+                    // two is wrong.
+                    $paramVarName = $param['var'];
+                }
+
+                if ($realName !== $paramVarName) {
                     $code = 'ParamNameNoMatch';
                     $data = [
                         $param['var'],
```

Comment:
```suggestion
                        $paramVarName,
```

Extra test to see the issue & safeguard the fix:
```php
 * @param string &$case A string passed in by reference with a case mismatch.
...
public function variablesPassedByReference(&$foo, &$bar, $baz, $qux, &$CASE)
```

Reply:
If I were making these changes, I'd keep the "param has leading & while not passed by ref" vs the "param name mismatch" checks completely separate as these are effectively checking two completely different things.

So, yes, I'd do the name compare in a way to disregard a potential leading `&` completely.